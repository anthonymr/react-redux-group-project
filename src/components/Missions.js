import { useSelector } from 'react-redux';
import Mission from './Mission';
import style from '../styles/Missions.module.css';

function Missions() {
  const { missions } = useSelector((state) => state.missions);

  return (
    <section>
      <table className={style.missionstable}>
        <thead className={style.missionstablehead}>
          <tr className={style.missionstablerow}>
            <th scope="col" style={{ width: '10%' }}>Mission</th>
            <th scope="col" style={{ width: '60%' }}>Description </th>
            <th scope="col" style={{ width: '15%' }}>Status</th>
            <th scope="col" style={{ width: '15%' }}>Join</th>
          </tr>
        </thead>
        { missions.map((mission) => (
          <tbody className={style.tablebody} key={mission.id}>
            <Mission
              name={mission.name}
              description={mission.description}
              id={mission.id}
              reserved={mission?.reserved}
            />
          </tbody>
        ))}
      </table>
    </section>
  );
}

export default Missions;
