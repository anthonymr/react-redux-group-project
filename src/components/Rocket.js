import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import { bookRocket, cancelBookRocket } from '../redux/rockets/rocketsSlice';
import styles from '../styles/Rocket.module.css';

function Rocket({
  id, flickrImages, name, description, reserved,
}) {
  const dispatch = useDispatch();

  const reserveClickHandler = () => {
    dispatch(bookRocket(id));
  };

  const cancelationClickHandler = () => {
    dispatch(cancelBookRocket(id));
  };

  return (
    <article className={styles.rocketContainer}>
      <div className={styles.rocketPicture}>
        <img src={flickrImages[0]} alt={name} />
      </div>
      <div className={styles.rocketContent}>
        <h2>{name}</h2>
        <p>
          { reserved && <span className={styles.reservedLabel}>Reserved</span> }
          {description}
        </p>
        { !reserved && <button type="button" onClick={reserveClickHandler}>Reserve Rocket</button> }
        { reserved
          && (
          <button type="button" onClick={cancelationClickHandler} className={styles.cancel}>
            Cancel Reservation
          </button>
          )}
      </div>
    </article>
  );
}

Rocket.propTypes = {
  id: PropTypes.string.isRequired,
  flickrImages: PropTypes.instanceOf(Array).isRequired,
  name: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  reserved: PropTypes.bool,
};

Rocket.defaultProps = {
  reserved: false,
};

export default Rocket;
