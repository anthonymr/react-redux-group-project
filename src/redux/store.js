import { configureStore } from '@reduxjs/toolkit';
import rocketsReducer from './rockets/rocketsSlice';
import missionsReducer from './missions/missionSlice';

const store = configureStore({
  reducer: {
    missions: missionsReducer,
    rockets: rocketsReducer,
  },
});

export default store;
