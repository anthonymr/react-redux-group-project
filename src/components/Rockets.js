import { useSelector } from 'react-redux';
import Rocket from './Rocket';

function Rockets() {
  const { rockets } = useSelector((state) => state.rockets);

  return (
    <section>
      <ul>
        {
          rockets.map((rocket) => (
            <li key={rocket.id}>
              <Rocket flickrImages={rocket.flickr_images} name={rocket.name} description={rocket.description} />
            </li>
          ))
        }
      </ul>
    </section>
  );
}

export default Rockets;
