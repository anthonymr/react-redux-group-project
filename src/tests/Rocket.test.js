import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/store';
import Missions from '../components/Missions';
import Rockets from '../components/Rockets';
import Rocket from '../components/Rocket';

it('renders correctly', () => {
  const mockedRocket = {
    id: 'testingRocket',
    flickr_images: ['https://imgur.com/DaCfMsj.jpg'],
    name: 'testingRocket',
    description: 'testingDescription',
    reserved: false,
  };
  const tree = renderer
    .create(
      <Provider store={store}>
        <Missions />
        <Rockets />
        <Rocket
          id={mockedRocket.id}
          flickrImages={mockedRocket.flickr_images}
          name={mockedRocket.name}
          description={mockedRocket.description}
          reserved={mockedRocket?.reserved}
        />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
