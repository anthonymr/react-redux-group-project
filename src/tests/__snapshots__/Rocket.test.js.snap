// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <section>
    <table
      className="missionstable"
    >
      <thead
        className="missionstablehead"
      >
        <tr
          className="missionstablerow"
        >
          <th
            scope="col"
            style={
              Object {
                "width": "10%",
              }
            }
          >
            Mission
          </th>
          <th
            scope="col"
            style={
              Object {
                "width": "60%",
              }
            }
          >
            Description 
          </th>
          <th
            scope="col"
            style={
              Object {
                "width": "15%",
              }
            }
          >
            Status
          </th>
          <th
            scope="col"
            style={
              Object {
                "width": "15%",
              }
            }
          >
            Join
          </th>
        </tr>
      </thead>
    </table>
  </section>,
  <section
    className="rocketsContainer"
  >
    <ul
      className="rocketList"
    />
  </section>,
  <article
    className="rocketContainer"
  >
    <div
      className="rocketPicture"
    >
      <img
        alt="testingRocket"
        src="https://imgur.com/DaCfMsj.jpg"
      />
    </div>
    <div
      className="rocketContent"
    >
      <h2>
        testingRocket
      </h2>
      <p>
        testingDescription
      </p>
      <button
        onClick={[Function]}
        type="button"
      >
        Reserve Rocket
      </button>
    </div>
  </article>,
]
`;
